src/horizon_client/horizon_client.rs

```diff
@@ -60,7 +59,7 @@ impl HorizonClient {
     /// [GET /accounts/{account_id}](https://www.stellar.org/developers/horizon/reference/endpoints/accounts-single.html)
     pub async fn get_single_account(
         &self,
-        request: &SingleAccountRequest,
+        request: &SingleAccountRequest<AccountId>,
```

Comment:
Why is there a difference between the `&impl ValidAccountsRequest` and `&SingleAccountRequest<AccountId>` in both signatures (looking at `get_account_list`)?